Improve windows performance

Gatsik opened the pull request:
* Migrate to GitHub Actions
* Do not query `Win32_VideoController` information from the `WMI`, which can be very slow, and use `GetSystemMetrics()` functions instead -- this also requires at least Windows Vista

~~The part with generating and passing `UID_PUBKEY_BYTES` remains unclear~~

Resolves #15

Gatsik committed:
Reference UnicodeString in a way compiler doesn't complain

Gatsik committed:
Replace deprecated FastWriter with Json::StreamBuilder

Gatsik committed:
Get screen dimensions faster

by using GetSystemMetrics function
querying Win32_VideoController from the WMI can take a lot of time

Gatsik committed:
Migrate to GitHub Actions

Gatsik committed:
Make cryptopp work

* include cryptopp headers with quotes otherwise
compiler on Linux can't see them
* use different cryptopp types for windows and linux
because MXE-built cryptopp version doesn't support CryptoPP:byte
and make refuses to build old cryptopp version on without errors

Gatsik committed:
Do not add extension to archive name when uploading artifacts

remove unnecessary mkdir call

Gatsik commented:
@muellni @Brutus5000 ping

Brutus5000 commented:
The uid secret is made available in the secret context as `UID_PUBLIC_KEY_PEM`. I guess this needs to be incorporated into the Github workflow somehow? Echo it into a file or something?

Commit:
Pass to cmake UID_PUBKEY_BYTES

which are generated from the pubkey file
which is created from the repository secret
in the environment 'release'

Brutus5000 reviewed (approved):
I guess there is only one way to figure out if it works :D
